### ISSUE TITLE
Viewport: Editor can crash when running automated tests that configure the Viewport layout

### ISSUE THREAD

**Issue by jckand-amzn** [o3de/o3de#9276]
**Describe the bug**
Editor can crash when running automated tests that configure the Viewport layout. This crash has only been observed once in AR (https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/stabilization%2F2205/40/pipeline), and has not been reproduced locally.

**Prerequisites**
1. Editor is configured and built to run AutomatedTesting.
2. All AutomatedTesting assets are processed.

**Steps to reproduce** LOW REPRO RATE
1. Run the Dynamic Vegetation Main suite:

```
python\python.cmd -m pytest --build-directory build\bin\profile .\AutomatedTesting\Gem\PythonTests\largeworlds\dyn_veg\TestSuite_Main.py
```

**Expected behavior**
Suite runs without issue.

**Actual behavior**
An Editor crash can rarely occur when executing the test test_LayerSpawner_InstancesRefreshUsingCorrectViewportCamera, which uses 
```
azlmbr.legacy.general.set_view_pane_layout(1)
```
to set a 2 Viewport configuration.

**Call Stack**
```
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  ==================================================================
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:15> (System) - Trace::Assert
[2022-04-28T18:15:48.007Z] E   [editor_test.log]   D:/workspace/o3de/Code/Framework/AzCore/AzCore/Memory/HphaSchema.cpp(1333): (17624) 'void __cdecl AZ::HpAllocator::bucket_free_direct(void *,unsigned int)'
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:15> (System) - HPHA Assert, expression: "bi == p->bucket_index()"
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:15> (System) - ------------------------------------------------
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Code\Framework\AzCore\AzCore\Memory\HphaSchema.cpp (1336) : AZ::HpAllocator::bucket_free_direct
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Code\Framework\AzCore\AzCore\Memory\HphaSchema.cpp (2523) : AZ::HphaSchema::DeAllocate
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassSystem.cpp (197) : AZ::RPI::PassSystem::BuildPasses
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassSystem.cpp (298) : AZ::RPI::PassSystem::ProcessQueuedChanges
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\RenderPipeline.cpp (323) : AZ::RPI::RenderPipeline::OnPassModified
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\RenderPipeline.cpp (397) : AZ::RPI::RenderPipeline::OnStartFrame
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Scene.cpp (695) : AZ::RPI::Scene::PrepareRender
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\RPISystem.cpp (305) : AZ::RPI::RPISystem::RenderTick
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::SystemTickEvents,AZ::SystemTickEvents,0,1>::Dispatcher<AZ::EBus<AZ::SystemTickEvents,AZ::SystemTickEvents> >::Broadcast<void (__cdecl AZ:
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1357) : AZ::ComponentApplication::TickSystem
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Code\Editor\CryEdit.cpp (2290) : CCryEditApp::IdleProcessing
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Code\Editor\CryEdit.cpp (2176) : CCryEditApp::OnIdle
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - D:\workspace\o3de\Code\Editor\Core\QtEditorApplication.cpp (312) : Editor::EditorQtApplication::maybeProcessIdle
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC2ED7F72A (Qt5Core) : QTimer::timeout
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC2ED782DD (Qt5Core) : QObject::event
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC33B149A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC33B13A23 (Qt5Widgets) : QApplication::notify
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC2ED517CB (Qt5Core) : QCoreApplication::notifyInternal2
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC2ED9C1B9 (Qt5Core) : QEventDispatcherWin32Private::sendTimerEvent
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC2ED9A5E9 (Qt5Core) : QEventDispatcherWin32::event
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC33B149A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC33B13A23 (Qt5Widgets) : QApplication::notify
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC2ED517CB (Qt5Core) : QCoreApplication::notifyInternal2
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC2ED53575 (Qt5Core) : QCoreApplicationPrivate::sendPostedEvents
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - 00007FFC317F37BF (qwindows) : qt_plugin_query_metadata
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - ==================================================================
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (System) - ====Assert added to ignore list by spec and verbosity setting.====
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:16> (Exit) - Exception with exit code: 0xc0000005
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Code\Legacy\CrySystem\DebugCallStack.cpp (223) : DebugCallStack::handleException
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - 00007FFC42CB7B6C (KERNELBASE) : UnhandledExceptionFilter
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - 00007FFC45DE8440 (ntdll) : memset
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - 00007FFC45DCFDD6 (ntdll) : _C_specific_handler
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - 00007FFC45DE4A2F (ntdll) : _chkstk
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - 00007FFC45D44CEF (ntdll) : RtlWalkFrameChain
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - 00007FFC45DE379E (ntdll) : KiUserExceptionDispatcher
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Code\Framework\AzCore\AzCore\std\hash_table.h (1079) : AZStd::hash_table<AZStd::Internal::UnorderedSetTableTraits<AZ::RHI::Resource *,AZStd::hash<AZ::RHI::Resource *>,AZStd::equal_to<AZ::RHI::Resource *>,AZStd::allocator,0> >::insert_im
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RHI\Code\Source\RHI\ResourcePool.cpp (187) : AZ::RHI::ResourcePool::InitResource
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RHI\Code\Source\RHI\ShaderResourceGroupPool.cpp (67) : AZ::RHI::ShaderResourceGroupPool::InitGroup
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Shader\ShaderResourceGroup.cpp (93) : AZ::RPI::ShaderResourceGroup::Init
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Shader\ShaderResourceGroup.cpp (67) : AZ::RPI::ShaderResourceGroup::CreateInternal
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (172) : AZStd::Internal::function_util::get_invoker<AZStd::intrusive_ptr<AZ::RPI::ShaderResourceGroup> __cdecl(AZ::Data::AssetData *,AZStd::any const *),AZStd::Internal::func
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Code\Framework\AtomCore\AtomCore\Instance\InstanceDatabase.h (344) : AZ::Data::InstanceDatabase<AZ::RPI::ShaderResourceGroup>::FindOrCreate
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Shader\ShaderResourceGroup.cpp (56) : AZ::RPI::ShaderResourceGroup::Create
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ComputePass.cpp (95) : AZ::RPI::ComputePass::LoadShader
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ComputePass.cpp (48) : AZ::RPI::ComputePass::ComputePass
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\Feature\Common\Code\Source\CoreLights\LightCullingRemap.cpp (38) : AZ::Render::LightCullingRemap::Create
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (84) : AZStd::Internal::function_util::get_invoker<AZStd::intrusive_ptr<AZ::RPI::Pass> __cdecl(AZ::RPI::PassDescriptor const &),AZStd::Internal::function_util::function_ptr_t
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassFactory.cpp (106) : AZ::RPI::PassFactory::CreatePassFromIndex
[2022-04-28T18:15:48.007Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassFactory.cpp (160) : AZ::RPI::PassFactory::CreatePassFromRequest
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassSystem.cpp (444) : AZ::RPI::PassSystem::CreatePassFromRequest
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ParentPass.cpp (343) : AZ::RPI::ParentPass::BuildInternal
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\Pass.cpp (1237) : AZ::RPI::Pass::Build
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> (Exit) - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ParentPass.cpp (347) : AZ::RPI::ParentPass::BuildInternal
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - 00007FFC42CB7B6C (KERNELBASE) : UnhandledExceptionFilter
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - 00007FFC45DE8440 (ntdll) : memset
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - 00007FFC45DCFDD6 (ntdll) : _C_specific_handler
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - 00007FFC45DE4A2F (ntdll) : _chkstk
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - 00007FFC45D44CEF (ntdll) : RtlWalkFrameChain
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - 00007FFC45DE379E (ntdll) : KiUserExceptionDispatcher
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Code\Framework\AzCore\AzCore\std\hash_table.h (1079) : AZStd::hash_table<AZStd::Internal::UnorderedSetTableTraits<AZ::RHI::Resource *,AZStd::hash<AZ::RHI::Resource *>,AZStd::equal_to<AZ::RHI::Resource *>,AZStd::allocator,0> >::insert_im
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RHI\Code\Source\RHI\ResourcePool.cpp (187) : AZ::RHI::ResourcePool::InitResource
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RHI\Code\Source\RHI\ShaderResourceGroupPool.cpp (67) : AZ::RHI::ShaderResourceGroupPool::InitGroup
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Shader\ShaderResourceGroup.cpp (93) : AZ::RPI::ShaderResourceGroup::Init
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Shader\ShaderResourceGroup.cpp (67) : AZ::RPI::ShaderResourceGroup::CreateInternal
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (172) : AZStd::Internal::function_util::get_invoker<AZStd::intrusive_ptr<AZ::RPI::ShaderResourceGroup> __cdecl(AZ::Data::AssetData *,AZStd::any const *),AZStd::Internal::func
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Code\Framework\AtomCore\AtomCore\Instance\InstanceDatabase.h (344) : AZ::Data::InstanceDatabase<AZ::RPI::ShaderResourceGroup>::FindOrCreate
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Shader\ShaderResourceGroup.cpp (56) : AZ::RPI::ShaderResourceGroup::Create
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ComputePass.cpp (95) : AZ::RPI::ComputePass::LoadShader
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ComputePass.cpp (48) : AZ::RPI::ComputePass::ComputePass
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\Feature\Common\Code\Source\CoreLights\LightCullingRemap.cpp (38) : AZ::Render::LightCullingRemap::Create
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (84) : AZStd::Internal::function_util::get_invoker<AZStd::intrusive_ptr<AZ::RPI::Pass> __cdecl(AZ::RPI::PassDescriptor const &),AZStd::Internal::function_util::function_ptr_t
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassFactory.cpp (106) : AZ::RPI::PassFactory::CreatePassFromIndex
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassFactory.cpp (160) : AZ::RPI::PassFactory::CreatePassFromRequest
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\PassSystem.cpp (444) : AZ::RPI::PassSystem::CreatePassFromRequest
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ParentPass.cpp (343) : AZ::RPI::ParentPass::BuildInternal
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\Pass.cpp (1237) : AZ::RPI::Pass::Build
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\ParentPass.cpp (347) : AZ::RPI::ParentPass::BuildInternal
[2022-04-28T18:15:48.008Z] E   [editor_test.log]  <18:15:32> () - D:\workspace\o3de\Gems\Atom\RPI\Code\Source\RPI.Public\Pass\Pass.cpp (1237) : AZ::RPI::Pass::Build
```
**Found in Branch**
stabilization/2205@7731dc6a3ef371e5200a6fadecc35493d9667c98

**Comment by jckand-amzn** [comment]
Have been unsuccessful attempting to repro this manually, either by running the tests locally or manually configuring multiple viewports. Logging this for tracking, but without more information, there may not be much to do here.

**Comment by superkitcath** [comment]
This issue should have been fixed in the stabilisation branch; through https://github.com/o3de/o3de/issues/9107. Can you confirm?  @jackalbe

**Comment by jckand-amzn** [comment]
Have not encountered this again on an AR run, and have not been able to reproduce the issue locally. Appears to be resolved on stabilization/2205@a073665f466701bd946c9366b569e0e32a63c550.
